server/src/main/scala/org/elasticmq/server/config/ElasticMQServerConfig.scala

```diff
@@ -9,6 +9,7 @@ import org.elasticmq.server.QueueSorter
 import org.elasticmq.util.Logging
 
 class ElasticMQServerConfig(config: Config) extends Logging {
```

Comment:
This class only contains logic which parses configuration when the server is started & passes it to the actors, which manage the queues. The place where queues are managed & created is `QueueManagerActor` - and that's probably where name validation should go to.

This would probably become visible if you try to create a queue programatically - not through the config file, but through the SQS API. The config file is there to make it easier to setup tests, it's not the main interface for queue creation.

Try adding a test to `AmazonJavaSdkTestSuite` which tries to create a queue without the proper suffix, it will probably succeed (to create the queue, which means the test should fail :) ).

Reply:
I have added it into the `ElasticMQServerConfig` because from the docs it seems, that FIFO queue without the ".fifo" suffix is something, that should not exist. Thus I have made an assumption, that it should be reported to user as fast as it is possible and abort the startup of the server (due to incorrect config file).

`AmazonJavaSdkTestSuite` already has a test `FIFO queues should return an error if the queue's name does not end in .fifo`, which is creating a FIFO queue without the ".fifo" suffix and the outcome is `AmazonServiceException`. So, Amazon Java SDK is not allowing to create a FIFO queue without the ".fifo" suffix.

Reply:
But the configured queues are then created (synchronously) in `createQueues`, so any exceptions that happen during queue creation should be reported there anyway, no?

It seems that the Amazon Java SDK has a client-side check for this, but I would implement this in `QueueManagerActor` in case other clients (including non-official) don't have this check.

Reply:
Yeah, indeed it is better place for this validation. I have moved the queue name validation to `QueueManagerActor` and made small refactor of code related to SQS limits